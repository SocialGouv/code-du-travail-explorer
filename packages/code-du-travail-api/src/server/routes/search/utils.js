// merge by one of each

const merge = (res1, res2, max_result) => {
  //  simple array concat
  // return [...res1.hits, ...res2.hits].sort(sortByScore);

  // interleave results
  const res = [];
  Array.from({
    length: max_result
  }).forEach((x, i) => {
    if (res1.length > i) res.push(res1[i]);
    if (res2.length > i) res.push(res2[i]);
  });
  return res;
};

// Remove Duplicates

const removeDuplicate = arr =>
  Object.values(
    arr.reduce(
      (values, current) => ({
        ...values,
        [current._source.source + "/" + current._source.slug]: current
      }),
      {}
    )
  );

const mergeDuplicate = arr => {
  return Object.values(
    arr.reduce((acc, current) => {
      const unique = current._source.source + "/" + current._source.slug;
      // case we've already seen the same result from a different algo
      if (
        acc[unique] != undefined &&
        acc[unique]._source.algo != current._source.algo
      ) {
        acc[unique]._source.algo = "both";
      } else {
        acc[unique] = current;
      }
      return acc;
    }, {})
  );
};

const mergePipe = (a, b, max_result) => {
  const res = merge(a, b, a.length + b.length);
  return mergeDuplicate(res).slice(0, max_result);
};

exports.merge = merge;
exports.removeDuplicate = removeDuplicate;
exports.mergePipe = mergePipe;
