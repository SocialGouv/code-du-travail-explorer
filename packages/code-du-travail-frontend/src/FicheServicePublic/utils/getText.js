export default function getText(element = { text: "" }, joint = " ") {
  if (element.type === "text") {
    return element.$.trim();
  }
  if (element.$) {
    return element.$.map(child => getText(child, joint)).join(joint);
  }
  return "";
}
