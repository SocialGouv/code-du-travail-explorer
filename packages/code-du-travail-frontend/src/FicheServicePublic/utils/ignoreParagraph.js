const ignoreParagraph = children =>
  children.map(child => {
    if (child.name === "Paragraphe") {
      return child.$;
    }
    return child;
  });

export default ignoreParagraph;
