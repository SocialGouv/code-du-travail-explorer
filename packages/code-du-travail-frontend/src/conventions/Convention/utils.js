import { matopush } from "../../piwik";

export function trackAccordionPanelState(conventionName, eventName) {
  let previousPanelState = [];
  return function (currentPanelState) {
    if (currentPanelState.length > previousPanelState.length) {
      const lastOpenedPanel = currentPanelState.filter(
        (panel) => !previousPanelState.includes(panel)
      );
      matopush(["trackEvent", eventName, conventionName, lastOpenedPanel[0]]);
    }
    previousPanelState = currentPanelState;
  };
}
