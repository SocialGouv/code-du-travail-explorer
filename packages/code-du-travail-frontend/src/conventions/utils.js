export const sortByIntOrdre = (a, b) => {
  if (a.intOrdre < b.intOrdre) {
    return -1;
  }
  if (a.intOrdre > b.intOrdre) {
    return 1;
  }
  return 0;
};
