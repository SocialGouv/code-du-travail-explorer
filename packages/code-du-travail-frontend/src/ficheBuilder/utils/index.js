export { default as ignoreParagraph } from "./ignoreParagraph";
export { default as getText } from "./getText";
