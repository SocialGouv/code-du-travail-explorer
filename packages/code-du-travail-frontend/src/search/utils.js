import { getRouteBySource, SOURCES } from "@cdt/sources";

import { matopush } from "../piwik";

export const summarize = (text = "") =>
  text.length > 160 ? text.slice(0, text.indexOf(" ", 160)) + "â€¦" : text;

export const formatUrlMatomo = (source, slug, url) => {
  const trackedUrl =
    source === SOURCES.EXTERNALS ? url : `/${getRouteBySource(source)}/${slug}`;

  return trackedUrl;
};

export const reportSelectionToMatomo = (source, slug, url, algo) => {
  const qualifiedCall = JSON.stringify({
    algo,
    url: formatUrlMatomo(source, slug, url),
  });

  matopush(["trackEvent", "selectResult", qualifiedCall]);
};
