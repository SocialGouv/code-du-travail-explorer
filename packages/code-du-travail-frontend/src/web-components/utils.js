export const displayInViewport = (target, rootBoundaries) => {
  const viewportHeight =
    window.innerHeight || document.documentElement.clientHeight;
  const viewportWidth =
    window.innerWidth || document.documentElement.clientWidth;

  target.style.display = "block";
  target.style.top = rootBoundaries.height + "px";

  let targetBounding = target.getBoundingClientRect();

  target.style.left =
    -Math.round((targetBounding.width - rootBoundaries.width) / 2) + "px";

  targetBounding = target.getBoundingClientRect();
  // we are overflowing on the right
  if (targetBounding.right > viewportWidth) {
    target.style.right = "0px";
    target.style.left = "auto";
  }
  // we are overflowing on the left
  if (targetBounding.left < 0) {
    target.style.left = "0px";
    target.style.right = "auto";
  }
  // we are overflwing to the bottom
  if (targetBounding.bottom > viewportHeight) {
    target.style.top = -Math.round(targetBounding.height) + "px"; // don't make it go below header
  }
  // second pass to make sure the first pass did not
  // make the tooltip overflow on the opposite side :o
  targetBounding = target.getBoundingClientRect();
  if (targetBounding.left < 0) {
    target.style.right = targetBounding.left - 10 + "px";
  }
  if (targetBounding.right > viewportWidth) {
    target.style.left = viewportWidth - targetBounding.right - 10 + "px";
  }
  // don't make it go below header
  if (target.getBoundingClientRect().top < 120) {
    target.style.top = rootBoundaries.height + "px";
  }
};

const throttler = function (fn, interval) {
  let ready = true;
  return function (...args) {
    if (!ready) return;
    if (ready) {
      ready = false;
      window.setTimeout(() => {
        ready = true;
      }, interval);
      return fn(...args);
    }
  };
};

export const throttledDisplayInViewport = throttler(displayInViewport, 10);
