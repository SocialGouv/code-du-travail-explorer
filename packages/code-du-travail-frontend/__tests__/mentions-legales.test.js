import React from "react";
import { render } from "../test/utils";
import MentionLegales from "../pages/mentions-legales.js";

describe("<MentionLegales />", () => {
  it("should render", () => {
    const { container } = render(<MentionLegales />);
    expect(container).toMatchSnapshot();
  });
});
