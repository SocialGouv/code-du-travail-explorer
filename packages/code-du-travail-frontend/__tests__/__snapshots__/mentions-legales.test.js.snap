// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MentionLegales /> should render 1`] = `
.c12:not(:focus):not(:active) {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  padding: 0;
  overflow: hidden;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  border: 0;
}

.c37 {
  margin: 0;
  padding: 0;
  list-style-type: none;
}

.c1 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 1.25rem;
}

.c1 > *:first-child {
  margin-top: 0;
}

.c1 > *:last-child {
  margin-bottom: 0;
}

.c17 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 1.25rem;
  max-width: 46.25rem;
}

.c17 > *:first-child {
  margin-top: 0;
}

.c17 > *:last-child {
  margin-bottom: 0;
}

.c30 {
  max-width: 1200px;
  margin: 0 auto;
  padding: 0 1.25rem;
  padding: 0;
  max-width: 46.25rem;
}

.c30 > *:first-child {
  margin-top: 0;
}

.c30 > *:last-child {
  margin-bottom: 0;
}

.c16 {
  padding: 1.25rem 0;
}

.c18 {
  padding: 0.625rem 1.25rem;
  color: #434956;
  border: 1px solid #c9d3df;
  border-radius: 0.25rem;
  background-color: #fff;
}

.c18 > *:first-child {
  margin-top: 0;
}

.c18 > *:last-child {
  margin-bottom: 0;
}

.c25 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  display: inline-block;
  position: relative;
  width: 24px;
  height: 12px;
}

.c25::after,
.c25::before {
  display: block;
  position: absolute;
  top: 50%;
  width: 10px;
  height: 2px;
  background-color: currentColor;
  content: "";
}

.c25::before {
  left: 4px;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

[aria-expanded="true"] .c25::before,
[aria-selected="true"] .c25::before {
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

.c25::after {
  right: 4px;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
}

[aria-expanded="true"] .c25::after,
[aria-selected="true"] .c25::after {
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.c25::before,
.c25::after {
  -webkit-transition: -webkit-transform 0.25s ease,-webkit-transform 0.25s ease;
  -webkit-transition: transform 0.25s ease,-webkit-transform 0.25s ease;
  transition: transform 0.25s ease,-webkit-transform 0.25s ease;
}

.c23 + .c22 {
  border-top: 1px solid #c9d3df;
}

.c24 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c24:hover,
.c24:focus,
.c24:focus-within,
.c24[aria-expanded="true"] {
  color: #005994;
}

.c26 {
  padding: 1rem;
  -webkit-animation: bcCCNc 0.35s ease-in;
  animation: bcCCNc 0.35s ease-in;
}

.c26 > *:first-child {
  margin-top: 0;
}

.c26 > *:last-child {
  margin-bottom: 0;
}

.c26.accordion__body--hidden {
  display: none;
}

.c19 {
  margin-bottom: 1rem;
  padding: 0.625rem 1.25rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: #0c0c0e;
  background: #fee5ad;
  border-color: #fdd57b;
}

.c19 > *:first-child {
  margin-top: 0;
}

.c19 > *:last-child {
  margin-bottom: 0;
}

.c21 {
  margin-bottom: 1rem;
  padding: 0.625rem 1.25rem;
  border: 1px solid transparent;
  border-radius: 0.25rem;
  color: #0c0c0e;
  background: #e5f1fe;
  border-color: #b4d7fc;
}

.c21 > *:first-child {
  margin-top: 0;
}

.c21 > *:last-child {
  margin-bottom: 0;
}

.c32 {
  padding: 0.625rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border-style: solid;
  border-width: 1px 1px 2px 1px;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  color: #fff;
  background: #005994;
  border-color: #005994;
  border-bottom-color: #003a61;
}

.c32:not([disabled]):hover,
.c32:not([disabled]):focus {
  background: #0068ae;
  color: #fff;
}

.c32:not([disabled]):active {
  color: #fff;
  background: #0078c7;
  border-width: 2px 1px 1px 1px;
  border-color: #005994;
  outline: none;
}

.c32[aria-pressed="true"] {
  color: #fff;
  background: #0068ae;
  border-width: 2px 1px 1px 1px;
  border-color: #005994;
  border-top-color: #003a61;
  box-shadow: inset 0 1px 2px 0 #003a61;
}

.c32[aria-pressed="true"]:not([disabled]):hover,
.c32[aria-pressed="true"]:not([disabled]):focus {
  background-color: #005994;
}

.c32[aria-pressed="true"]:not([disabled]):active {
  border-top-color: #003a61;
}

.c32[disabled] {
  cursor: not-allowed;
  color: rgba(255,255,255,0.4);
}

.c7 {
  display: inline-block;
  padding: 0.5rem 0.625rem;
  font-size: 0.875rem;
  font-weight: bold;
  text-align: center;
  white-space: nowrap;
  line-height: 1rem;
  border-radius: 0.25rem;
  color: #0c0c0e;
  background: #e5f1fe;
}

.c11 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  text-align: left;
}

.c14 ul {
  z-index: 100;
}

.c14 li[role="option"]:nth-child(2n + 1) {
  background: #f9f9fc;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  margin: 0 auto;
  padding: 0;
  overflow: visible;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 0;
  padding: 0 calc(50px + 0.625rem) 0 1rem;
  width: 100%;
  height: 50px;
  font-size: inherit;
  font-family: inherit;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: #f9f9fc;
  border: 1px solid #c9d3df;
  border-radius: 0.25rem;
  -webkit-transition: border 250ms ease;
  transition: border 250ms ease;
}

.c13:focus {
  border-color: #006be6;
  outline: none;
}

.c15 {
  padding: 0.625rem 1rem;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  text-align: center;
  line-height: inherit;
  font-size: 1rem;
  font-weight: 600;
  vertical-align: middle;
  border-style: solid;
  border-width: 1px 1px 2px 1px;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: background-color 250ms ease;
  transition: background-color 250ms ease;
  padding: 1rem;
  color: #434956;
  line-height: 0;
  border: none;
  position: absolute;
  right: 1px;
  top: 1px;
  height: calc(50px - 2px);
  width: calc(50px - 2px);
  border-top-right-radius: 0.25rem;
  border-bottom-right-radius: 0.25rem;
}

.c15:hover {
  color: #8c94a6;
}

.c15:active {
  position: relative;
  top: 1px;
}

.c0 {
  background: #fff;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 1.25rem;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c4:link {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c4:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c5 {
  width: 5rem;
  height: 100%;
  margin-right: 1rem;
}

.c6 {
  font-size: 1.125rem;
  line-height: 1.4;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8 {
  margin-left: 1.25rem;
}

.c9 {
  width: 350px;
}

.c29 {
  text-align: center;
}

.c31 {
  margin-bottom: 1.25rem;
  text-align: left;
}

.c33 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c35 + .c34 {
  padding-left: 1rem;
}

.c36 {
  font-size: 1rem;
  font-weight: bold;
}

.c38 {
  margin: 0.5rem 0;
}

.c38 a,
.c38 a:visited {
  color: #434956;
}

.c20 {
  display: block;
  margin: 1.25rem 0;
}

.c27 {
  margin: 0;
  padding: 0;
  list-style-type: none;
  margin: 1.25rem 0;
  padding-left: 1.25rem;
  list-style-type: disc;
}

@media print {
  .c1 {
    max-width: 100%;
    padding: 0;
  }
}

@media print {
  .c17 {
    max-width: 100%;
    padding: 0;
  }
}

@media print {
  .c30 {
    max-width: 100%;
    padding: 0;
  }
}

@media (max-width:600px) {
  .c18 {
    padding: 0.625rem 1.25rem;
  }
}

@media print {
  .c18 {
    border: none;
    padding: 0 5pt;
  }
}

@media (max-width:600px) {
  .c10 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    height: auto;
  }
}

@media (max-width:600px) {
  .c13 {
    padding: 0 1rem;
  }
}

@media (max-width:980px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:600px) {
  .c2 {
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }
}

@media (max-width:980px) {
  .c9 {
    margin-top: 1.25rem;
    width: 70%;
  }
}

@media (max-width:600px) {
  .c9 {
    margin-top: 1.25rem;
    width: 100%;
  }
}

@media print {
  .c28 {
    display: none;
  }
}

@media (max-width:600px) {
  .c33 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:600px) {
  .c35 + .c34 {
    padding-left: 0;
  }
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1 c2"
    >
      <div
        class="c3"
      >
        <a
          class="c4"
          href="/"
          title="Code du travail numérique - retour à l'accueil"
        >
          <img
            alt="symbole de la Marianne, site officiel du gouvernement"
            class="c5"
            src="/static/assets/img/marianne.svg"
          />
          <span
            class="c6"
          >
            Code du travail
            <br />
            numérique
          </span>
        </a>
        <div
          class="c7 c8"
        >
          Version Bêta
        </div>
      </div>
      <div
        class="c9"
      >
        <form
          class="c10"
        >
          <label
            class="c11"
            for="main-search-input"
          >
            <span
              class="c12"
            >
              Rechercher
            </span>
            <div
              aria-expanded="false"
              aria-haspopup="listbox"
              aria-owns="react-autowhatever-cdtn-documents-suggester"
              role="combobox"
              style="flex: 1;"
            >
              <input
                aria-autocomplete="list"
                aria-controls="react-autowhatever-cdtn-documents-suggester"
                autocomplete="off"
                class="c13"
                id="main-search-input"
                name="query"
                placeholder="Recherche"
                type="search"
                value=""
              />
              <div
                class="c14"
                id="react-autowhatever-cdtn-documents-suggester"
                role="listbox"
              />
            </div>
          </label>
          <button
            aria-pressed="false"
            class="c15"
            type="submit"
          >
            <svg
              viewBox="0 0 20 20"
            >
              <title />
              <path
                d="M17.876 8.939c0 2.197-.819 4.208-2.123 5.773l4.023 4.022a.72.72 0 0 1 0 1.043.735.735 0 0 1-.522.223.736.736 0 0 1-.52-.223l-4.023-4.023a8.813 8.813 0 0 1-5.772 2.123C4.022 17.877 0 13.855 0 8.939 0 4.022 4.023 0 8.939 0c4.915 0 8.937 4.023 8.937 8.939zM9.001 16.5c4.124 0 7.499-3.375 7.499-7.5S13.125 1.5 9 1.5C4.877 1.5 1.5 4.875 1.5 9s3.376 7.5 7.5 7.5z"
                fill="currentColor"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </form>
      </div>
    </div>
  </header>
  <main
    id="main"
  >
    <div
      class="c16"
    >
      <div
        class="c17"
      >
        <div
          class="c18"
        >
          <h1>
            Mentions légales
          </h1>
          <div
            class="c19"
          >
            Ce site est en cours de construction, la fiabilité des réponses qui s’y trouvent ne sont pas garanties. L’ouverture officielle du site est prévue pour 2020.
          </div>
          <p>
            Le site est édité par l’
            <a
              href="https://incubateur.social.gouv.fr/"
            >
              Incubateur des ministères sociaux
            </a>
             
            situé :
          </p>
          <address
            class="c20"
          >
            Tour Mirabeau
            <br />
            39-43 Quai André Citroën
            <br />
            75015 PARIS
            <br />
            <br />
            Tél. : 01 40 56 60 00
          </address>
          <h3>
            Directeur de la publication
          </h3>
          <p>
            Hélène BRISSET
          </p>
          <h3>
            Hébergement
          </h3>
          <p>
            Ce site est hébergé en propre par le Ministère des Affaires sociales et de la Santé :
          </p>
          <address
            class="c20"
          >
            Ministère des affaires sociales et de la santé
            <br />
            14 avenue Duquesne
            <br />
            75530 PARIS
          </address>
          <div
            class="c21"
          >
            <p>
              Le
               
              <a
                href="https://github.com/SocialGouv/code-du-travail-numerique"
              >
                code du logiciel
              </a>
               
              est libre, et peut donc être vérifié et amélioré par toutes et tous à l’adresse suivante:
               
              <a
                href="https://github.com/SocialGouv/code-du-travail-numerique"
              >
                https://github.com/SocialGouv/code-du-travail-numerique
              </a>
              .
            </p>
          </div>
          <div
            class="accordion"
            data-accordion-component="Accordion"
          >
            <div
              class="c22 c23"
              data-accordion-component="AccordionItem"
            >
              <div
                aria-level="3"
                class="accordion__heading"
                data-accordion-component="AccordionItemHeading"
                role="heading"
              >
                <div
                  aria-controls="accordion__panel-0"
                  aria-disabled="false"
                  aria-expanded="false"
                  class="c24"
                  data-accordion-component="AccordionItemButton"
                  id="accordion__heading-0"
                  role="button"
                  tabindex="0"
                >
                  <h2>
                    Modalités d’utilisation
                  </h2>
                  <div
                    class="c25"
                    role="presentation"
                  />
                </div>
              </div>
              <div
                aria-hidden="true"
                aria-labelledby="accordion__heading-0"
                class="c26"
                data-accordion-component="AccordionItemPanel"
                hidden=""
                id="accordion__panel-0"
              >
                <h3>
                  Traitement des données à caractère personnel
                </h3>
                <p>
                  Le code du travail numérique ne vous demande ni ne stocke d’information nominative.
                </p>
                <p>
                  Pour autant, nous enregistrons les informations saisies dans la barre de recherche. Elles sont conservées pendant deux années pour analyser les usages, améliorer la précision des réponses apportées et améliorer le service.
                </p>
                <p>
                  Nous nous engageons à ne jamais céder ces informations à des tiers.
                </p>
                <p>
                  Vous avez un droit d’accès, de rectification et de suppression de vos données. Pour l’exercer, faites-nous parvenir une demande en précisant la date et l’heure précise de la requête - ces éléments sont indispensables pour nous permettre de retrouver votre recherche - par voie électronique à l’adresse suivante :
                   
                  <a
                    href="mailto:codedutravail@beta.gouv.fr"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    codedutravail@beta.gouv.fr
                  </a>
                   
                  ou par voie postale :
                </p>
                <address
                  class="c20"
                >
                  Direction des systèmes d’information
                  <br />
                  Ministère des affaires sociales et de la santé
                  <br />
                  39-43 Quai André Citroën
                  <br />
                  75015 PARIS
                </address>
                <h3>
                  Utilisation de témoins de connexion (« cookies »)
                </h3>
                <p>
                  Nous collectons également des données par l’intermédiaire de dispositifs appelés “cookies” permettant d’établir des mesures statistiques de fréquentation et d’utilisation du site pouvant être utilisées à des fins de suivi et d’amélioration du service :
                </p>
                <ul
                  class="c27"
                >
                  <li
                    class=""
                  >
                    Les données collectées ne sont pas recoupées avec d’autres traitements.
                  </li>
                  <li
                    class=""
                  >
                    Le cookie déposé sert uniquement à la production de statistiques anonymes.
                  </li>
                  <li
                    class=""
                  >
                    Le cookie ne permet pas de suivre la navigation de l’internaute sur d’autres sites.
                  </li>
                </ul>
                <p>
                  La mesure d’audience (nombre de visites, pages consultées) est réalisée par un outil libre intitulé Matomo spécifiquement paramétré, respectant les conditions d’exemption du consentement de l’internaute définies par la recommandation « Cookies » de la Commission nationale informatique et libertés (CNIL).
                </p>
                <p>
                  Pour la période de construction, ce site utilise l’outil Hotjar, notamment pour obtenir les retours écrits des utilisateurs concernant la pertinence des résultats proposés dans le cadre de leurs recherches. Hotjar utilise des cookies et d’autres technologies pour collecter des informations sur l’équipement et les comportements des visiteurs. Cet outil stocke ces informations dans un profil utilisateur pseudonymisé. Ni Hotjar ni l’éditeur n’utiliseront ces informations pour identifier des utilisateurs individuels ou pour les associer à d’autres données. Vous pouvez consulter la politique de confidentialité de
                   
                  <a
                    href="https://www.hotjar.com/legal/policies/privacy"
                  >
                    Hotjar
                  </a>
                   . Ces données ne seront jamais cédées. La durée de conservation de ces informations ne saurait dépasser deux ans.
                </p>
                <p>
                  Vous pouvez librement désactiver ce service, sur l’ensemble des sites internet susceptibles de l’utiliser en suivant la procédure présentée
                   
                  <a
                    href="https://www.hotjar.com/legal/compliance/opt-out"
                  >
                    ici: https://www.hotjar.com/legal/compliance/opt-out
                  </a>
                  .
                </p>
                <p>
                  Le site utilise également la solution
                   
                  <a
                    href="https://polyfill.io/v3/"
                  >
                    Polyfill.io
                  </a>
                  , ce service permet de faire fonctionner l’ensemble des fonctionnalités du code du travail numérique sur tous les navigateurs, quelless que soient leurs générations.
                </p>
                <p>
                  À tout moment, vous pouvez refuser l’utilisation des cookies et désactiver le dépôt sur votre ordinateur en utilisant la fonction dédiée de votre navigateur (fonction disponible notamment sur Microsoft Internet Explorer 11, Google Chrome, Mozilla Firefox, Apple Safari et Opera).
                </p>
                <p>
                  L’ensemble de ces informations sont nécessaires à la mise en oeuvre de ce service public numérique.
                </p>
              </div>
            </div>
            <div
              class="c22 c23"
              data-accordion-component="AccordionItem"
            >
              <div
                aria-level="3"
                class="accordion__heading"
                data-accordion-component="AccordionItemHeading"
                role="heading"
              >
                <div
                  aria-controls="accordion__panel-1"
                  aria-disabled="false"
                  aria-expanded="false"
                  class="c24"
                  data-accordion-component="AccordionItemButton"
                  id="accordion__heading-1"
                  role="button"
                  tabindex="0"
                >
                  <h2>
                    Accessibilité
                  </h2>
                  <div
                    class="c25"
                    role="presentation"
                  />
                </div>
              </div>
              <div
                aria-hidden="true"
                aria-labelledby="accordion__heading-1"
                class="c26"
                data-accordion-component="AccordionItemPanel"
                hidden=""
                id="accordion__panel-1"
              >
                <p>
                  La conformité aux normes d’accessibilité numérique est un objectif ultérieur mais nous tâchons de rendre dès la conception, ce site accessible à toutes et à tous.
                </p>
                <h3>
                  Signaler un dysfonctionnement
                </h3>
                <p>
                  Si vous rencontrez un défaut d’accessibilité vous empêchant d’accéder à un contenu ou une fonctionnalité du site, merci de nous en faire part.
                </p>
                <p>
                  Si vous n’obtenez pas de réponse rapide de notre part, vous êtes en droit de faire parvenir vos doléances ou une demande de saisine au Défenseur des droits.
                </p>
                <h3>
                  En savoir plus
                </h3>
                <p>
                  Pour en savoir plus sur la politique d’accessibilité numérique de l’État :
                   
                  <a
                    href="http://references.modernisation.gouv.fr/accessibilite-numerique"
                  >
                    http://references.modernisation.gouv.fr/accessibilite-numerique
                  </a>
                </p>
              </div>
            </div>
            <div
              class="c22 c23"
              data-accordion-component="AccordionItem"
            >
              <div
                aria-level="3"
                class="accordion__heading"
                data-accordion-component="AccordionItemHeading"
                role="heading"
              >
                <div
                  aria-controls="accordion__panel-2"
                  aria-disabled="false"
                  aria-expanded="false"
                  class="c24"
                  data-accordion-component="AccordionItemButton"
                  id="accordion__heading-2"
                  role="button"
                  tabindex="0"
                >
                  <h2>
                    Sécurité
                  </h2>
                  <div
                    class="c25"
                    role="presentation"
                  />
                </div>
              </div>
              <div
                aria-hidden="true"
                aria-labelledby="accordion__heading-2"
                class="c26"
                data-accordion-component="AccordionItemPanel"
                hidden=""
                id="accordion__panel-2"
              >
                <p>
                  Le site est protégé par un certificat électronique, matérialisé pour la grande majorité des navigateurs par un cadenas. Cette protection participe à la confidentialité des échanges.
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
  <footer
    class="c28"
  >
    <div
      class="c16"
    >
      <div
        class="c1 c29"
      >
        <h2>
          Besoin d’un accompagnement personnalisé ?
        </h2>
        <div
          class="c30 c31"
        >
          Les services de renseignement en droit du travail peuvent vous donner des informations juridiques générales relatives au Code du travail, aux conventions collectives, à la jurisprudence. Ils peuvent également vous conseiller et vous orienter dans vos démarches.
        </div>
        <button
          aria-pressed="false"
          class="c32"
        >
          Contacter les services de renseignement
        </button>
      </div>
    </div>
    <div
      class="c16"
    >
      <div
        class="c1 c33"
      >
        <div
          class="c34 c35"
        >
          <h3
            class="c36"
          >
            Code du travail numérique
          </h3>
          <ul
            class="c37"
          >
            <li
              class="c38"
            >
              <a
                href="/droit-du-travail"
              >
                Le droit du travail
              </a>
            </li>
            <li
              class="c38"
            >
              <a
                href="/glossaire"
              >
                Glossaire
              </a>
            </li>
            <li
              class="c38"
            >
              <a
                href="/a-propos"
              >
                À propos
              </a>
            </li>
            <li
              class="c38"
            >
              <a
                href="/mentions-legales"
              >
                Mentions légales
              </a>
            </li>
            <li
              class="c38"
            >
              <a
                href="mailto:codedutravail@beta.gouv.fr"
              >
                Contact
              </a>
            </li>
          </ul>
        </div>
        <div
          class="c34 c35"
        >
          <h3
            class="c36"
          >
            Aidez-nous à améliorer cet outil
          </h3>
          <ul
            class="c37"
          >
            <li
              class="c38"
            >
              <a
                href="https://github.com/SocialGouv/code-du-travail-numerique/tree/vX.Y.Z"
                rel="noopener noreferrer"
                target="_blank"
              >
                Contribuer sur Github
              </a>
            </li>
            <li
              class="c38"
            >
              <a
                href="https://github.com/SocialGouv/code-du-travail-numerique/releases/tag/vX.Y.Z"
                rel="noopener noreferrer"
                target="_blank"
              >
                Journal des modifications
              </a>
            </li>
          </ul>
        </div>
        <div
          class="c34 c35"
        >
          <h3
            class="c36"
          >
            En collaboration avec
          </h3>
          <ul
            class="c37"
          >
            <li
              class="c38"
            >
              <a
                href="https://travail-emploi.gouv.fr/ministere/organisation/article/dgt-direction-generale-du-travail"
                rel="noopener noreferrer"
                target="_blank"
              >
                La Direction Générale du Travail
              </a>
            </li>
            <li
              class="c38"
            >
              <a
                href="https://incubateur.social.gouv.fr/"
                rel="noopener noreferrer"
                target="_blank"
              >
                L’incubateur des ministères sociaux
              </a>
            </li>
            <li
              class="c38"
            >
              <a
                href="https://beta.gouv.fr/"
                rel="noopener noreferrer"
                target="_blank"
              >
                beta.gouv.fr
              </a>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
